### PR TITLE
Jax interface for all devices.

### PR DESCRIPTION
**Pull request by Thenerdstation**
**Context:**
Previously, only the `default.qubit.jax` device supported the JAX interface. This PR adds JAX interface support to all devices.

**Description of the Change:**
The recent addition of `jax.experimental.host_callback` allows us to finally do `jax -> numpy -> numpy -> jax` within a `jax.jit` function! This PR adds the needed scaffolding to do it. 

**Benefits:**
Everyone can start using JAX more and I won't take any more execuses.

**Possible Drawbacks:**
None.

**Related GitHub Issues:**
#943 

**TODOs**

* Gradient support
* Vmap support

### CODE REVIEW

**Comment by Thenerdstation** [comment]
The build is being weird, but it does pass
